api/include/opentelemetry/logs/event_logger.h

```diff
@@ -0,0 +1,115 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#ifdef ENABLE_LOGS_PREVIEW
+
+#  include <chrono>
+#  include <map>
+#  include <vector>
+
+#  include "opentelemetry/common/attribute_value.h"
+#  include "opentelemetry/common/key_value_iterable.h"
+#  include "opentelemetry/common/key_value_iterable_view.h"
+#  include "opentelemetry/common/macros.h"
+#  include "opentelemetry/common/timestamp.h"
+#  include "opentelemetry/logs/log_record.h"
+#  include "opentelemetry/logs/logger.h"
+#  include "opentelemetry/logs/logger_type_traits.h"
+#  include "opentelemetry/logs/severity.h"
+#  include "opentelemetry/nostd/shared_ptr.h"
+#  include "opentelemetry/nostd/span.h"
+#  include "opentelemetry/nostd/string_view.h"
+#  include "opentelemetry/nostd/type_traits.h"
+#  include "opentelemetry/nostd/unique_ptr.h"
+#  include "opentelemetry/trace/span_context.h"
+#  include "opentelemetry/trace/span_id.h"
+#  include "opentelemetry/trace/trace_flags.h"
+#  include "opentelemetry/trace/trace_id.h"
+#  include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace logs
+{
+/**
+ * Handles event log record creation.
+ **/
+class EventLogger
+{
+public:
+  virtual ~EventLogger() = default;
+
+  /* Returns the name of the logger */
+  virtual const nostd::string_view GetName() noexcept = 0;
+
+  /* Returns the delegate logger of this event logger */
+  virtual nostd::shared_ptr<Logger> GetDelegateLogger() noexcept = 0;
+
+  /**
+   * Create a Log Record object
+   *
+   * @return nostd::unique_ptr<LogRecord>
+   */
+  virtual nostd::unique_ptr<LogRecord> CreateLogRecord() noexcept = 0;
+
+  /**
+   * Emit a event Log Record object
+   *
+   * @param event_name Event name
+   * @param log_record Log record
+   */
+  virtual void EmitEvent(nostd::string_view event_name,
+                         nostd::unique_ptr<LogRecord> &&log_record) noexcept = 0;
+
+  /**
+   * Emit a event Log Record object with arguments
+   *
+   * @param event_name Event name
+   * @param log_record Log record
+   * @tparam args Arguments
+   */
+  template <class... ArgumentType>
+  void EmitEvent(nostd::string_view event_name,
+                 nostd::unique_ptr<LogRecord> &&log_record,
+                 ArgumentType &&... args)
```

Comment:
We are passing log_record, and parameter packs both ? Isn't event_name and log_record object enough.

Reply:
You are right, I remove it now.